test/datalog_tests/test-stream.sh

```diff
@@ -86,6 +88,29 @@ run_rel_query_test() {
 
 }
 
+run_count_unique_test() {
+    echo Running unique user count test $1 for $2 batches of $3 events;
+
+    (
+    for (( i=1; i<=$2; i++ ))
+    do
+        echo "echo Transaction $i;"
+        echo "start;"
+        for (( j=1; j<=$3; j++ ))
+        do
+            echo "insert UserSession$1($(($i * $2 + $j))),"
+            #echo "insert UserSession$1($j),"
+        done
+        echo "commit dump_changes;"
+    done
+    ) > unique_bench$1.dat
+    /usr/bin/time ./streams_ddlog/target/release/streams_cli -w 1 --no-store < unique_bench$1.dat > unique_bench$1.dump
+    diff -q unique_bench$1.dump unique_bench$1.dump.expected
```

Comment:
How much faster is it?

Reply:
The version using hashsets is 20x faster than the equivalent implementation using regular sets.  The gap growth with the size of the set. I also had an optimized set-based implementation that was only 5 times slower.